lib/zhuyin.ex

```diff
@@ -0,0 +1,326 @@
+    "ㄦ" => "er",
+    "ㄢ" => "an",
+  }
+  @finals %{
+    "ㄧ" => "i",
+    "ㄨ" => "u",
+    "ㄩ" => "v",
+    "ㄚ" => "a",
+    "ㄛ" => "o",
+    "ㄜ" => "e",
+    "ㄝ" => "e",
+    "ㄞ" => "ai",
+    "ㄟ" => "ei",
+    "ㄠ" => "ao",
+    "ㄡ" => "ou",
+    "ㄢ" => "an",
+    "ㄣ" => "en",
+    "ㄤ" => "ang",
+    "ㄥ" => "eng",
+    "ㄦ" => "er",
+    "ㄧㄚ" => "ia",
+    "ㄨㄚ" => "ua",
+    "ㄧㄥ" => "ing",
+    "ㄧㄤ" => "iang",
+    "ㄧㄝ" => "ie",
+    "ㄨㄛ" => "uo",
+    "ㄨㄥ" => "ong",
+    "ㄨㄤ" => "uang",
+    "ㄧㄠ" => "iao",
+    "ㄨㄞ" => "uai",
+    "ㄩㄝ" => "ve",
+    "ㄩㄥ" => "iong",
+    "ㄧㄡ" => "iu",
+    "ㄨㄟ" => "ui",
+    "ㄧㄢ" => "ian",
+    "ㄨㄢ" => "uan",
+    "ㄩㄢ" => "van",
+    "ㄧㄣ" => "in",
+    "ㄨㄣ" => "un",
+    "ㄩㄣ" => "vn",
+  }
+
+  @doc"""
+  Convert zhuyin to pinyin
+
+  # Examples
+   iex>  decode_zhuyin("ㄝ")
+   None
+
+   iex>  decode_zhuyin("ㄐㄩ˙")
+   "jv"
+
+   iex>  decode_zhuyin("ㄌㄩˇ")
+   "lv3"
+
+   iex>  decode_zhuyin("ㄓㄨㄢˋ")
+   "zhuan4"
+
+   iex>  decode_zhuyin("ㄓㄠˊ")
+   "zhao2"
+
+   iex>  decode_zhuyin("ㄓˋ")
+   "zhi4"
+
+   iex>  decode_zhuyin("ㄌㄥ")
+   "leng1"
+
+   iex>  decode_zhuyin("ㄕㄨㄟˇ")
+   "shui3"
+
+   iex>  decode_zhuyin("ㄌㄧㄡˊ")
+   "liu2"
+
+   iex>  decode_zhuyin("ㄧˊ")
+   "yi2"
+
+   iex>  decode_zhuyin("ㄇㄧㄣˊ")
+   "min2"
+
+   iex>  decode_zhuyin("ㄐˇ")
+   None
+
+   // Zhuyin specific
+   iex>  decode_zhuyin("ㄨㄥˊ")
+   "weng2"
+
+   iex>  decode_zhuyin("ㄨˊ")
+   "wu2"
+
+   iex>  decode_zhuyin("ㄩ")
+   "yu1"
+
+   iex>  decode_zhuyin("ㄩㄝˇ")
+   "yue3"
+
+   iex>  decode_zhuyin("ㄩㄝˋ")
+   "yue4"
+
+   iex>  decode_zhuyin("ㄩㄢˊ")
+   "yuan2"
+
+   iex>  decode_zhuyin("ㄌㄩㄢˋ")
+   "lvan4"
+
+   iex>  decode_zhuyin("ㄌㄢˇ")
+   "lan3"
+
+   iex>  decode_zhuyin("ㄦˊ")
+   "er2"
+
+   iex>  decode_zhuyin("a5")
+   None
+
+   // Special case
+   iex>  decode_zhuyin("ㄦ˙")
+   "r"
+  """
+  # @spec decode_zhuyin(String.t()) :: Integer.t()
+   def decode_zhuyin(input) do
+    case input |> Pinyin.ZhuYinParsers.zhuyin_word |> IO.inspect do
+      {:ok, parsed, _rest, _other1, _other2, _other3} ->
+        first = parsed |> List.first
+        if first == input do
+          None
+        else
+          first |> Zhuyin.to_pinyin |> Pinyin.numbered
+        end
+      true -> None
+    end
+   end
+
+  @spec to_pinyin(t()) :: Pinyin.t()
+  # Special case for this final and tone combination
+  def to_pinyin(%Zhuyin{initial: "", final: "ㄦ", tone: 0}) do
+    Pinyin.create("", "r", 0)
+  end
+  def to_pinyin(zhuyin = %Zhuyin{}) do
+    pinyin_initial = if zhuyin.initial == "" do
+      ""
+    else
+      @initials[zhuyin.initial]
+    end
+    pinyin_final = if zhuyin.initial == "" do
+      @standalone_finals[zhuyin.final]
+    else
+      @finals[zhuyin.final]
+    end
+    Pinyin.create(pinyin_initial, pinyin_final, zhuyin.tone)
+  end
+  def to_pinyin(list) when is_list(list) do
+    list |> Enum.map(&Zhuyin.to_pinyin/1)
+  end
+
+  @spec to_zhuyin(Pinyin.t() | Pinyin.pinyin_list()) :: t()
```

Comment:
Would it make sense to put this in the `Pinyin` module? If not I'd name this `from_pinyin` instead.

Reply:
> If not I'd name this from_pinyin instead.

I prefer this, done.